app/scripts/modules/core/src/managed/StatusBubble.tsx

```diff
@@ -36,16 +38,47 @@ const transitionConfig = {
   config: { mass: 1, tension: 400, friction: 30 },
 };
 
-export const StatusBubble = memo(({ iconName, appearance, size }: IStatusBubbleProps) => {
+export const StatusBubble = memo(({ appearance, iconName, size, quantity }: IStatusBubbleProps) => {
   const transitions = useTransition(iconName, null, transitionConfig);
+  let quantityPill: ReactNode = null;
+  let withQuantity = false;
+
+  // Render quantity text only for size >= small.
+  if (quantity && (size === 'small' || size === 'medium' || size === 'large' || size === 'extraLarge')) {
+    withQuantity = true;
+    quantityPill = (
+      <span
+        className={classnames({
+          Pill: true,
+          [size]: true,
+          [`status-${appearance}`]: true,
+        })}
+      >
+        {quantity}
+      </span>
+    );
+  }
 
   return (
-    <div className={`StatusBubble status-bubble-${appearance} sp-padding-${paddingBySize[size]}`}>
+    <>
```

Comment:
have you tried this out live w/ changing icon names? i can't tell from reading the code but i'm sort of wondering whether the fact that mapping over `transitions` will sometimes generate 2 side-by-side divs here might lead to layout jank during transitions. the grid layout stuff that was previously on `StatusBubble` (now on `IconWrapper`) was originally to make sure that transitioning elements sat atop each other vs. getting stacked vertically or horizontally without constraining them to specific heights/widths

Reply:
Good question. I'm going to debug this further to see if this is happening

Reply:
Working on this now and this PR could be merged once I confirm if it is not an issue.

Reply:
There is an issue, I'm working on fixing it now.

Reply:
Fixed it and ready for merge.

![animating-status-bubble](https://user-images.githubusercontent.com/357832/80838945-00f7d100-8baf-11ea-806b-e5cd20a2b2dc.gif)